ambry-replication/src/main/java/com.github.ambry.replication/RemoteReplicaInfo.java

```diff
@@ -170,7 +179,20 @@ synchronized void onTokenPersisted() {
 
   @Override
   public String toString() {
-    return replicaId.toString();
+    return replicaId.getPartitionId() + ":" + replicaId.getDataNodeId() + ":" + replicaId.getReplicaPath();
+  }
+
+  @Override
+  public int compareTo(RemoteReplicaInfo other) {
+    return this.getReplicaId()
+        .getPartitionId()
+        .toPathString()
+        .compareTo(other.getReplicaId().getPartitionId().toPathString());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return toString().equals(obj.toString());
```

Comment:
I wonder why we didn't check the class type of `obj` here. Feel like we are supposed to check as follows:
```
    if (this == obj) {
      return true;
    }
    if (obj == null || getClass() != obj.getClass()) {
      return false;
    }
    RemoteReplicaInfo info = (RemoteReplicaInfo) obj;
    return compareTo(info) == 0;
```

Reply:
That's some stale code for test. Will address it.